src/frontend/serialization_phase.nit

```diff
@@ -56,6 +61,34 @@ redef class ADefinition
 	end
 end
 
+private class SerializationPhaseRename
+	super Phase
+
+	redef fun process_annotated_node(node, nat)
+	do
+		var text = nat.n_atid.n_id.text
+		if text != "serialize_as" then return
+
+		if not node isa AAttrPropdef then
+			toolcontext.error(node.location,
+				"Syntax Error: annotation `{text}` applies only to attributes.")
+			return
+		end
+
+		# Can't use `arg_as_string` because it needs the literal phase
+		var args = nat.n_args
+		if args.length != 1 or not args.first isa AStringFormExpr then
```

Comment:
It tried that, bu if I'm not mistaken `arg_as_string` needs the literal phase. This new phase creates a lot of literals so it runs before the literal phase.